### PR TITLE
ripple effect

### PR DESCRIPTION
**Pull request by bso-odoo**

### CODE REVIEW

File: addons/website/static/src/interactions/ripple_effect.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import { registry } from "@web/core/registry";
+import { Interaction } from "@website/core/interaction";
+
+export class RippleEffect extends Interaction {
+    static selector = ".btn, .dropdown-toggle, .dropdown-item";
+    dynamicContent = {
+        "_root": {
+            "t-on-click": this.onClick,
```
Comment (ged-odoo) [inline addons/website/static/src/interactions/ripple_effect.js:8]: this maybe an opportunity to use a t-att-class:
```js
"t-att-class": () => ({"o_js_ripple_effect": this.isActive})
```

File: addons/website/static/src/interactions/ripple_effect.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        },
+    };
+    duration = 350;
+
```
Comment (ged-odoo) [inline addons/website/static/src/interactions/ripple_effect.js:15]:
```js
setup() {
  this.isActive = false;
  this.timeoutID = null;
  this.rippleEl = null;
} 
```

File: addons/website/static/src/interactions/ripple_effect.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    destroy() {
+        this.rippleEl?.remove();
+    }
+
+    /**
+     * @param {boolean} [toggle]
+     */
+    toggleRippleEffect(toggle) {
```
Comment (ged-odoo) [inline addons/website/static/src/interactions/ripple_effect.js]: remove

File: addons/website/static/src/interactions/ripple_effect.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    onClick(ev) {
+        if (!this.rippleEl) {
+            this.rippleEl = document.createElement("span");
+            this.rippleEl.classList.add("o_ripple_item");
+            this.rippleEl.style.animationDuration = `${this.duration}ms`;
+            this.el.appendChild(this.rippleEl);
+        }
+
+        clearTimeout(this.timeoutID);
+        this.toggleRippleEffect(false);
```
Comment (ged-odoo) [inline addons/website/static/src/interactions/ripple_effect.js]:
```js
this.isActive = false;
```

File: addons/website/static/src/interactions/ripple_effect.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        const diameter = Math.max(this.el.clientWidth, this.el.clientHeight);
+
+        this.rippleEl.style.width = `${diameter}px`;
+        this.rippleEl.style.height = `${diameter}px`;
+        this.rippleEl.style.top = `${ev.pageY - offsetY - diameter / 2}px`;
+        this.rippleEl.style.left = `${ev.pageX - offsetX - diameter / 2}px`;
+
+        this.toggleRippleEffect(true);
+        this.timeoutID = this.waitForTimeout(() => {
+            this.toggleRippleEffect(false);
```
Comment (ged-odoo) [inline addons/website/static/src/interactions/ripple_effect.js]: this.isActive = false;

File: addons/website/static/src/interactions/ripple_effect.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        // The diameter need to be recomputed because a change of window width
+        // can affect the size of a button (e.g. media queries).
+        const diameter = Math.max(this.el.clientWidth, this.el.clientHeight);
+
+        this.rippleEl.style.width = `${diameter}px`;
+        this.rippleEl.style.height = `${diameter}px`;
+        this.rippleEl.style.top = `${ev.pageY - offsetY - diameter / 2}px`;
+        this.rippleEl.style.left = `${ev.pageX - offsetX - diameter / 2}px`;
+
+        this.toggleRippleEffect(true);
```
Comment (ged-odoo) [inline addons/website/static/src/interactions/ripple_effect.js]: this.isActive = true;

File: addons/website/static/src/interactions/ripple_effect.js (hunk 5 above)
Comment (bso-odoo) [reply addons/website/static/src/interactions/ripple_effect.js]: Will require a synchronous `updateDOM`

**Pull request merged**